src/logger.c

```diff
@@ -103,7 +103,10 @@ static void log_file_init(struct log_file *f, char *name, int size)
 	    O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR)) < 0)
 		pexit("open: %s", path_expand(name));
 
-	f->ptr = f->buffer = mem_alloc(size + LINE_BUFFER_SIZE);
+	/* plain will now always be < LINE_BUFFER_SIZE. We add some extra bytes
+	 * so that there is ALWAYS enough buffer to write our line, and we no
+	 * longer have to check length before a write (.pot or .log file) */
+	f->ptr = f->buffer = mem_alloc(size + LINE_BUFFER_SIZE + 512);
```

Comment:
What are this "some extra bytes" for exactly, and what makes 512 a good figure? Is this actually the headroom for adding a field separator plus a cleartext to the ciphertext (that was truncated to LINE_BUFFER_SIZE)? In that case we might want to simply use PLAINTEXT_BUFFER_SIZE instead of 512.